.github/workflows/build-toolchain.yml

```diff
@@ -662,17 +827,17 @@ jobs:
 
   release:
     runs-on: ubuntu-latest
-    needs: [context, windows-build]
+    needs: [context, windows-build-x64]
     if: inputs.create_release == true
     steps:
       - uses: actions/download-artifact@v4
         with:
-          name: installer-amd64
+          name: installer-amd64-built-on-amd64
           path: ${{ github.workspace }}/tmp/amd64
 
       - uses: actions/download-artifact@v4
         with:
-          name: installer-arm64
+          name: installer-arm64-built-on-amd64
```

Comment:
We shouldn't be changing the name of the artifact, the native vs cross should generate the same content.

Reply:
The problem is that the upload-artifact fails and causes the entire job to fail if something with the same name has already been uploaded and the order between x64/arm64, which depends on the build progress speed, is unpredictable. Can you think of a better workaround?

Reply:
That shouldn't matter - we can download only the native builds (i.e. the amd64 installer from x64, the arm64 installer from arm64).

Reply:
Right; I'm saying we change that. Only the AMD64 build comes from the AMD64 build and only the ARM64 build comes from the ARM64 build.

Reply:
@Steelskin This is the discussion we had. It's about how to avoid artifact name conflicts between the cross-compiled arm64 build and the native arm64 build. Currently I use a suffix to avoid collisions. And I use a similar "...built on <os>/<arch>" suffix to the job names to make it less confusing on the GH workflow UI though job name conflicts don't cause a build failure. Appreciate your feedback!

Reply:
I think the main issue is that you are building the same artifacts with both an arm64 and an x64 host (see my other comment). Basically, only one host should be building for one given target. Anything more is redundant.

Reply:
As above and what compnerd and I discussed in other places, the premise is that we want to keep three builds: x64-built-on-x64, arm64-built-on-x64, arm64-built-on-arm64. They are redundant from the point of view of producing release binaries but they are not from the point of view of ensuring the three builds work.

Reply:
If we want to keep maintaining the arm64-built-on-x64 build then yes, we don't have a choice, we need the ability to differentiate between the 2 builds or we are going to hit a conflict with the artifact name.

Reply:
Gotcha

Reply:
@compnerd Are you okay with this suffix approach?